app/src/androidTest/java/ch/epfl/sweng/erpa/activities/DiceActivityTest.java

```diff
@@ -0,0 +1,206 @@
+import static android.support.test.espresso.action.ViewActions.clearText;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.Matchers.hasToString;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(AndroidJUnit4.class)
+public class DiceActivityTest {
+    @Rule
+    public final IntentsTestRule<DiceActivity> intentsTestRule = new IntentsTestRule<>(
+            DiceActivity.class);
+
+    @Test
+    public void testCorrectOutputForD4Die() {
+        onView(withId(R.id.d4_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(4)));
+        }
+    }
+
+    @Test
+    public void testCorrectOutputForD6Die() {
+        onView(withId(R.id.d6_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(6)));
+        }
+    }
+
+    @Test
+    public void testCorrectOutputForD8Die() {
+        onView(withId(R.id.d8_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(8)));
+        }
+    }
+
+    @Test
+    public void testCorrectOutputForD10Die() {
+        onView(withId(R.id.d6_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(10)));
+        }
+    }
+
+    @Test
+    public void testCorrectOutputForD12Die() {
+        onView(withId(R.id.d12_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(12)));
+        }
+    }
+
+    @Test
+    public void testCorrectOutputForD20Die() {
+        onView(withId(R.id.d20_number)).perform(typeText("1")).perform(closeSoftKeyboard());
+        for (int i = 0; i < 10; ++i) {
+            onView(withId(R.id.rollButton)).perform(click());
+            onView(withId(R.id.roll1)).check(matches(textViewHasCorrectValue(20)));
+        }
+    }
```

Comment:
Gotta agree with codeclimate. You should refactor this (since all you're changing is which view you're working with)